### PR TITLE
Moving common logic of StackFrame into shared

### PR DESCRIPTION
**Pull request by maryamariyan**
The CoreRT diff to can be viewed here: https://github.com/dotnet/corert/pull/5859

Related to: #9474

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Diagnostics/Stackframe.cs [hunk 1]
```diff
@@ -263,7 +267,17 @@ public override string ToString()
...
+                includeFileInfoIfAvailable = true;
+            }
+            else
+            {
+                includeFileInfoIfAvailable = AppendStackFrameWithoutMethodBase(sb);
+            }
 
+            if (includeFileInfoIfAvailable)
+            {
+#endif
```
Comment (maryamariyan) [inline src/System.Private.CoreLib/src/System/Diagnostics/Stackframe.cs]: @jkotas Please let me know if you prefer the way I included this #ifdef to be improved and whether or not you have suggestions for improving this part.

File: src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        /// Line number representing the current code location if available, 0 otherwise.
+        /// </summary>
+        private int _lineNumber;
+
+        /// <summary>
+        /// Column number representing the current code location if available, 0 otherwise.
+        /// </summary>
+        private int _columnNumber;
+
+#if CORECLR
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs]: This should be deleted instead.

File: src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        [System.Runtime.Serialization.OptionalField]
+#endif
+		/// <summary>
+        /// This flag is set to true when the frame represents a rethrow marker.
+        /// </summary>
+        private bool _isLastFrameFromForeignExceptionStackTrace;
+
+        internal void InitMembers()
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs]: Can this be private?

File: src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+                        else
+                            fFirstTyParam = false;
+
+                        sb.Append(typars[k].Name);
+                        k++;
+                    }
+
+                    sb.Append('>');
+                }
+#if !CORECLR
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs]: Would it be better to add `AppendStackFrameWithoutMethodBase` method to the CoreCLR specific part that just returns false?

Comment (maryamariyan) [reply src/System.Private.CoreLib/shared/System/Diagnostics/Stackframe.cs]: OK. Thanks, will do that.

File: src/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 5]
```diff
@@ -509,7 +509,7 @@
     <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Debugger.cs" />
     <Compile Include="$(BclSourcesRoot)\System\Diagnostics\ICustomDebuggerNotification.cs" />
     <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Stacktrace.cs" />
-    <Compile Include="$(BclSourcesRoot)\System\Diagnostics\Stackframe.cs" />
+    <Compile Include="$(BclSourcesRoot)\System\Diagnostics\StackFrame.cs" />
```
Comment (jkotas) [inline src/System.Private.CoreLib/System.Private.CoreLib.csproj]: Rename this to StackFrame.CoreCLR.cs ?

**Comment by jkotas** [comment]
@trylek I believe you have tried to do this some time ago, but then abandoned it.

**Comment by maryamariyan** [comment]
@jkotas please review the combination of the two diffs below which tries to remove extra ifdef by moving logic to non-shared files:

- corert: [here](https://github.com/dotnet/corert/pull/5859/commits/3bdd6b38590b4f194d51afaf467df84cd697b992)
- coreclr: [here](https://github.com/dotnet/coreclr/pull/18134/commits/f3f54294666810e8f41314b3d69e927c2ba81e04)

**Comment by jkotas** [comment]
Builds breaks ... also check my comment I have left on the CoreRT change.

**Comment by trylek** [comment]
When I was implementing the StackTrace / StackFrame API's in CoreRT about a year ago, I was intentionally trying to make it as close to CoreCLR as I could but for some reason it couldn't be 100% at that time, I must admit I no longer recall why, I have a vague recollection that it was maybe due to some differences in the StackTrace / StackFrame formatting but I'm not completely sure; on way or another, it was likely something easily solvable by splitting the file into a common part and a specific part. I was also out of time booked for this work, I believe I did get to something like 90% of the way towards unifying the code.

**Comment by jkotas** [comment]
Build break: `error CS2001: Source file '/mnt/resource/j/workspace/dotnet_coreclr/master/x64_debug_centos7.1_innerloop_prtest/src/System.Private.CoreLib/src/System/Diagnostics/StackFrame.cs' could not be found`

**Comment by jkotas** [comment]
LGTM otherwise

**Pull request merged**